src/subscription_manager/exceptions.py

```diff
@@ -118,6 +118,14 @@ def format_proxy_exception(self, exc: connection.ProxyException, _: str) -> str:
             )
         return PROXY_ADDRESS_MESSAGE.format(hostname=proxy_address)
 
+    def format_unknown_content(self, exc: connection.UnknownContentException, _: str) -> str:
+        content = exc.content
+        if content is not None:
+            return UNKNOWN_CONTENT_MESSAGE_CONTENT.format(
+                code=exc.code, title=exc.title, content=terminal_printable_content(content)
```

Comment:
> Printing HTML page of e.g. 404 page to the terminal is just nor professional. Probably nobody will read it and our customers will think about us that we are ...

Our customers already think that we are not professional because all they get is "something fails", and they are left wondering _what_ exactly failed (even before _why_).

Beside that, 404 is already handled in a different way, so it will not be printed this should the remote server (be it Candlepin or anything else in the middle) reply with 404.

`UnknownContentException`, and the handling of it improved by this work, is a very corner case situation in which the answer is not JSON, and the HTTP code is not a well-known one (see the apidoc of `UnknownContentException`). It is not even sure that users will get the same error reply by running the same command again. Because of this, I think it's critical to properly let the user know _what_ happened in that case, so there is a chance of doing something with it -- also, from my experience with customers, copy&paste of terminal/console output to support cases is one of the first things they will do.

> I propose to print to the terminal something like: "Unknown content is bytes long. More details can be found in rhsm.log" and then we can print this unknown content to `rhsm.log`.

One of the goals of this work is **exactly** to give less and less reasons to the users of subscription-manager to open `rhsm.log` to actually know _what_ exactly happened/failed.

Telling users "something failed, look in that file" adds a level of indirection that makes it more difficult for them to know the situation; plus, I also was told that "well, if you know what is the problem, why is not that printed directly rather than burying it in a log file?", which is honestly hard to disagree.

Also, having to look in the log file has some drawbacks:
- you need to know exactly the time a command that failed happened, and then you need to find it there
- the log file is not exactly user-friendly to read
- when running subscription-manager in non-interactive contexts (say from your own script/app, ansible, etc), you most likely will have only stdout+stderr of subscription-manager available

----

Considering the above, I still find it preferable to print the content in case of such corner case situations.

(Also, did not we not have already an internal discussion about this some weeks ago?)

Reply:
> Our customers already think that we are not professional because all they get is "something fails", and they are left wondering what exactly failed (even before why).

Well, justifying something wrong, because something else is wrong is just not good idea at all.

> Telling users "something failed, look in that file" adds a level of indirection that makes it more difficult for them to know the situation; plus, I also was told that "well, if you know what is the problem, why is not that printed directly rather than burying it in a log file?", which is honestly hard to disagree.

I'm sure that 99.99% of users are able to use `less`, `tail`, etc. and display end of the file `/var/log/rhsm/rhsm.log` in this corner case. If you will read this: https://clig.dev/, then of course will find justification for your opinion and my opinion too.

> Also, did not we not have already an internal discussion about this some weeks ago?

Yeah, we did and I didn't agree with printing HTML page to console during this meeting.

If you still do not agree with me, then I think we can go with your implementation and fix it later in better way.